src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -1562,21 +1617,35 @@ - (BOOL)pairDevice:(uint64_t)deviceID
       setupPINCode:(uint32_t)setupPINCode
              error:(NSError * __autoreleasing *)error
 {
+    // Track overall commissioning
+    MATTER_LOG_METRIC_BEGIN(kMetricDeviceCommissioning);
```

Comment:
So we use the same metric for "start PASE handshake" and "start commissioning"?  That seems a bit confusing.

Reply:
Also, I am not sure we are doing the corresponding METRIC_END on all the codepaths where things terminate with async failures (e.g. if commissioning discovery fails).